### PR TITLE
Unattended Running

### PR DESCRIPTION
**Pull request by shurick81**
#40

### CODE REVIEW

**Comment by shurick81** [comment]
@brianlala let me know if you think I need to improve anything more

**Comment by shurick81** [comment]
By the way I have now tested it with CI run in GitLab and it worked just fine with my changes

**Comment by shurick81** [comment]
hi @brianlala was I too late before summer time :)

**Comment by shurick81** [comment]
@brianlala, would you like to merge this in order to make this script possible to run unattended?

File: Scripts/AutoSPSourceBuilder.ps1 [hunk 1]
```diff
@@ -62,6 +62,8 @@
     If this switch is omitted, and no languages have been specified on the command line, languages are skipped entirely.
 .PARAMETER UseExistingLocalXML
     If you want to use a custom or pre-existing AutoSPSourceBuilder.XML file, or simply want to skip downloading the official one on-the-fly, use this switch parameter.
+.PARAMETER Unattend
```
Comment (brianlala) [inline Scripts/AutoSPSourceBuilder.ps1:65]: Could you please update the parameter name to "Unattended"?

File: Scripts/AutoSPSourceBuilder.ps1 [hunk 2]
```diff
@@ -62,6 +62,8 @@
...
+    Disables additional user interation so that command can be run in CI/CD pipelines
```
Comment (brianlala) [inline Scripts/AutoSPSourceBuilder.ps1:66]: Minor changes, "interation" should be **interaction** and replace "command" with "script"

File: Scripts/AutoSPSourceBuilder.ps1 [hunk 3]
```diff
@@ -278,7 +282,9 @@ If (!([Security.Principal.WindowsPrincipal] [Security.Principal.WindowsIdentity]
 {
     Write-Warning " - You should run this script under an elevated PowerShell prompt. Launch an elevated PowerShell prompt by right-clicking the PowerShell shortcut and selecting `"Run as Administrator`"."
     Write-Warning " - Running without elevation may cause certain things to fail, e.g. file extraction."
-    Pause -action "proceed if you are sure this is OK, or Ctrl-C to exit" -key "y"
+    if ( !$Unattend ) {
```
Comment (brianlala) [inline Scripts/AutoSPSourceBuilder.ps1:285]: Could you please follow the style convention here and elsewhere (open brace on new line, watch out for extra/trailing spaces, etc.)?

**Review by brianlala — CHANGES_REQUESTED** [review]
Looks good, and sorry for the delay in reviewing! Just see inline commands and watch for style.

**Comment by shurick81** [comment]
Thanks Brian, now everything looks good

**Review by brianlala — APPROVED** [review]

**Pull request merged**